pyccel/parser/semantic.py

```diff
@@ -1116,11 +1138,37 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign,ar
             if not arr_in_multirets:
                 self._ensure_target(rhs, d_lhs)
 
-            var = self.check_for_variable(name)
+            var = self.check_for_variable(lhs)
 
             # Variable not yet declared (hence array not yet allocated)
             if var is None:
 
+                if isinstance(lhs, DottedName):
+                    prefix_parts = lhs.name[:-1]
+                    syntactic_prefix = prefix_parts[0] if len(prefix_parts) == 1 else DottedName(*prefix_parts)
+                    prefix = self._visit(syntactic_prefix)
+                    class_def = prefix.cls_base
+                    if isinstance(class_def, ClassDef) and lhs.name[0] == 'self':
```

Comment:
Shouldn't this be:
```suggestion
                    if isinstance(class_def, ClassDef) and prefix.name == 'self':
```
?

Otherwise this would be called if the class contained an instance of another class. E.g:
```python
def __init__(self, p : Point, x : float):
     self._p = p
     self._p.x = x
```

Reply:
It will not do a change cause the `check_for_varibale` returns the variable in the exemple of `self._p.x` so it doesn't access the condition `if var == None` but it's fixed in https://github.com/pyccel/pyccel/pull/1484/commits/e7b19d20720e2c2800a7858162928c0aae6ea639 cause it's more readable.

Reply:
Ah you're right :+1: